fastestimator/network.py

```diff
@@ -181,3 +174,62 @@ def _reduce_loss(self, batch, global_batch_size, epoch_losses, warm_up):
                     "please make sure loss is element-wise loss"
             reduced_loss[loss_name] = tf.reduce_sum(element_wise_loss) / global_batch_size
         return reduced_loss
+
+
+def build(model_def, model_name, optimizer, loss_name):
+    """build keras model instance in FastEstimator
+
+    Args:
+        model_def (function): function definition of tf.keras model
+        model_name (str, list, tuple): model name(s)
+        optimizer (str, optimizer, list, tuple): optimizer(s)
+        loss_name (str, list, tuple): loss name(s)
+
+    Returns:
+        model: model(s) compiled by FastEstimator
+    """
+    convert_tuple = False
+    if get_num_devices() > 1:
+        distribute_strategy = tf.distribute.MirroredStrategy()
+    else:
+        distribute_strategy = None
+    with distribute_strategy.scope() if distribute_strategy else NonContext():
+        model = model_def()
+        if isinstance(model, (list, tuple)):
+            assert len(model) == len(model_name) == len(optimizer) == len(loss_name)
```

Comment:
we need to call to_list on model_name, optimizer, loss_name first. else could end up fooled with 4 models and loss_name="loss"

Reply:
fixed